astropy/cosmology/core.py

```diff
@@ -451,11 +461,62 @@ def is_flat(self):
         """Return `True`, the cosmology is flat."""
         return True
 
-    @property
     @abc.abstractmethod
     def equivalent_nonflat(self: _FlatCosmoT) -> _CosmoT:
         """Return the equivalent non-flat-class instance of this cosmology."""
 
+    def clone(self, *, meta: Optional[Mapping]=None, to_nonflat: bool=False, **kwargs):
+        """Returns a copy of this object with updated parameters, as specified.
+
+        This cannot be used to change the type of the cosmology, except for
+        changing to the non-flat version of this cosmology.
+
+        Parameters
+        ----------
+        meta : mapping or None (optional, keyword-only)
+            Metadata that will update the current metadata.
+        to_nonflat : bool, optional keyword-only
+            Whether to change to the non-flat version of this cosmology.
+        **kwargs
+            Cosmology parameter (and name) modifications.
+            If any parameter is changed and a new name is not given, the name
+            will be set to "[old name] (modified)".
+
+        Returns
+        -------
+        newcosmo : `~astropy.cosmology.Cosmology` subclass instance
+            A new instance of this class with updated parameters as specified.
+            If no modifications are requested, then a reference to this object
+            is returned instead of copy.
```

Comment:
Yes. ``cosmology.clone()`` will return ``self``. I guess this might be less ambiguous as ``if no arguments are given``

Reply:
The phrasing is clear enough as it is. The reason I asked was this:
https://github.com/astropy/astropy/blob/614450553023af31ac1adddab18628d8c02f1e77/astropy/cosmology/core.py#L202-L209
I didn't see how this could return a reference to `self`, but that's only because I didn't notice it is done elsewhere:
https://github.com/astropy/astropy/blob/614450553023af31ac1adddab18628d8c02f1e77/astropy/cosmology/core.py#L185-L186
So there's no problem here.